### PR TITLE
openthread: remove support for NCP and related tools

### PR DESCRIPTION
**Pull request by edmont**
Remove support for NCP architecture, making RCP the default.

Remove documentation about wpantund, Pyspinel
and vendor hooks.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

Depends on:

- [x] https://projecttools.nordicsemi.no/bitbucket/projects/NCS-TEST/repos/test-fw-nrfconnect-thread/pull-requests/585/overview

KRKNWK-13088

### CODE REVIEW

**Comment by rlubos** [comment]
Is this only us dropping the support, or is OT planning to drop NCP as well?

File: doc/nrf/ug_logging.rst [hunk 1]
```diff
@@ -120,14 +120,10 @@ The Spinel protocol can be configured as a logging backend using the following K
 * :kconfig:`CONFIG_LOG_BACKEND_SPINEL` - This option enables the Spinel logging backend.
 * :kconfig:`CONFIG_LOG_BACKEND_SPINEL_BUFFER_SIZE` - This option defines the size of buffer used for logger output.
 
-To communicate using the Spinel protocol and gather logs, you need one of the following tools:
+To communicate using the Spinel protocol and gather logs, you need one of the `Pyspinel`_ tool:
```
Comment (wiba-nordic) [inline doc/nrf/ug_logging.rst:123]:
```suggestion
To communicate using the Spinel protocol and gather logs, you need the `Pyspinel`_ tool:
```

File: doc/nrf/ug_thread_commissioning.rst [hunk 2]
```diff
@@ -236,7 +234,7 @@ One device will act as a Commissioner and the other will be a Joiner.
 Requirements
 ============
 
-To configure on-mesh Thread commissioning, you need at least two development kits that are compatible with either the CLI or the NCP samples.
+To configure on-mesh Thread commissioning, you need at least two development kits that are compatible with either the CLI or the Coprocessor samples.
```
Comment (wiba-nordic) [inline doc/nrf/ug_thread_commissioning.rst:237]:
```suggestion
To configure on-mesh Thread commissioning, you need at least two development kits that are compatible with either the CLI or the Co-processor samples.
```

File: doc/nrf/ug_thread_commissioning.rst [hunk 3]
```diff
@@ -259,7 +257,7 @@ This network needs to be manually disabled.
 Disabling the Thread network
 ============================
 
-The |NCS|'s Thread CLI and NCP samples come with the autostart feature, which means that the devices will form the network automatically without user intervention.
+The |NCS|'s Thread CLI and Coprocessor samples come with the autostart feature, which means that the devices will form the network automatically without user intervention.
```
Comment (wiba-nordic) [inline doc/nrf/ug_thread_commissioning.rst:260]:
```suggestion
The |NCS|'s Thread CLI and Co-processor samples come with the autostart feature, which means that the devices will form the network automatically without user intervention.
```

File: samples/openthread/coprocessor/README.rst [hunk 4]
```diff
@@ -8,7 +8,7 @@ Thread: Co-processor
    :depth: 2
 
 The :ref:`Thread <ug_thread>` Co-processor sample demonstrates how to implement OpenThread's :ref:`thread_architectures_designs_cp` inside the Zephyr environment.
-Depending on the configuration, the sample uses the :ref:`thread_architectures_designs_cp_ncp` architecture or :ref:`thread_architectures_designs_cp_rcp` architecture.
+Depending on the configuration, the sample uses the :ref:`thread_architectures_designs_cp_rcp` architecture.
```
Comment (wiba-nordic) [inline samples/openthread/coprocessor/README.rst:11]:
```suggestion
The sample uses the :ref:`thread_architectures_designs_cp_rcp` architecture.
```
Alternatively, if it is not mandatory, it's "The sample can be configured to use the :ref:`thread_architectures_designs_cp_rcp` architecture."

**Review by wiba-nordic — CHANGES_REQUESTED** [review]
A couple of small improvements.

**Review by lmaciejonczyk — COMMENTED** [review]
We should remove Pyspinel parts as it's tool for NCP only.

File: samples/openthread/coprocessor/README.rst [hunk 5]
```diff
@@ -81,11 +82,10 @@ User interface
 
 All the interactions with the application are handled using serial communication.
 
-You can interact with the sample through :ref:`ug_thread_tools_wpantund`, using ``wpanctl`` commands.
-If you use the RCP architecture (see :kconfig:`CONFIG_OPENTHREAD_COPROCESSOR_RCP`), you can alternatively use ``ot-daemon`` or ``ot-cli`` with commands listed in `OpenThread CLI Reference`_.
+You can interact with the sample through ``ot-daemon`` or ``ot-cli`` with commands listed in `OpenThread CLI Reference`_.
 See :ref:`ug_thread_tools_ot_apps` for more information.
 
-Both NCP and RCP support communication with the development kit using `Pyspinel`_ commands.
+Communication with the development kit it is also supported using `Pyspinel`_ commands.
```
Comment (wiba-nordic) [inline samples/openthread/coprocessor/README.rst:88]:
```suggestion
Communication with the development kit using `Pyspinel`_ commands is also supported.
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]
Noticed one more.

**Review by lmaciejonczyk — APPROVED** [review]

**Comment by canisLupus1313** [comment]
@edmont removing the `overlay-rcp.conf` causes the rcp samples to fail build stage there is also a need to change this in testing repo.

**Review by canisLupus1313 — CHANGES_REQUESTED** [review]

**Comment by edmont** [comment]
> @edmont removing the `overlay-rcp.conf` causes the rcp samples to fail build stage there is also a need to change this in testing repo.

Yes, the matching CI PR is linked in the description.

**Comment by edmont** [comment]
Test are passing for the updated test branch: https://jenkins-ncs.nordicsemi.no/job/latest/job/test-fw-nrfconnect-thread/job/debugging%252Fremove-ncp/4/

**Review by canisLupus1313 — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**